### PR TITLE
acts: new versions 23.[3-5].0, 24.0.0, 25.0.[0-1], 26.0.0, 27.[0-1].0, 28.0.0

### PR DESCRIPTION
**Pull request by wdconinc**
- https://github.com/acts-project/acts/compare/v23.2.1...v23.3.0:
  - `py-onnxruntime@1.12.0:` now required when `+onnx`, previous versions conflict with `1.13:` (https://github.com/acts-project/acts/pull/1835), support commented out to pass audit while awaiting #37064
- https://github.com/acts-project/acts/compare/v23.3.0...v23.4.0
  - `actsvg@0.4.29:` now required when `+svg`
- https://github.com/acts-project/acts/compare/v23.4.0...v23.5.0
  - `actsvg@0.4.30:` now required when `+svg`
- https://github.com/acts-project/acts/compare/v23.5.0...v24.0.0
  - nothing relevant
- https://github.com/acts-project/acts/compare/v24.0.0...v25.0.0
  - new variant `mlpack`, depends on new package `mlpack` (#38277)
  - `actsvg@0.4.33:` now required when `+svg`
- https://github.com/acts-project/acts/compare/v25.0.0...v26.0.0
  - new tests that check build failure invariants: support not added
- https://github.com/acts-project/acts/compare/v26.0.0...v27.0.0
  - new FPE monitoring plugin: support not added
- https://github.com/acts-project/acts/compare/v27.0.0...v27.1.0
  - nothing relevant
- https://github.com/acts-project/acts/compare/v27.1.0...v28.0.0
  - `actsvg@0.4.35:` now required when `+svg` (added, no changes required https://github.com/acts-project/actsvg/compare/v0.4.33...v0.4.35)

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the commits exist (in association with the versions).  Deferring merge decision to @HadrienG2 .

**Review by HadrienG2 — APPROVED** [review]

**Comment by wdconinc** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/acts/package.py [hunk 1]
```diff
@@ -213,7 +223,13 @@ class Acts(CMakePackage, CudaPackage):
...
         default=False,
         description="Build the EDM4hep examples",
-        when="@19.4.0: +examples",
+        when="@19.4.0:24 +examples",
+    )
+    variant(
+        "edm4hep",
+        default=False,
+        description="Build the EDM4hep examples",
+        when="@19.4.0:24 +examples",
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts/package.py]:
```suggestion
```

**Comment by wdconinc** [comment]
We're starting to run a bit behind in ACTS versions supported in spack. This aims to do a bit of catchup.

I've removed the blockers due to audit on not-yet-merged PRs, and added conflicts for those variants while the relevant PRs are merged. @HadrienG2 Could you give another review?

File: var/spack/repos/builtin/packages/acts/package.py [hunk 2]
```diff
@@ -271,12 +285,21 @@ class Acts(CMakePackage, CudaPackage):
     depends_on("hepmc3 @3.2.1:", when="+hepmc3")
     depends_on("heppdt", when="+hepmc3 @:4.0")
     depends_on("intel-tbb @2020.1:", when="+examples +tbb")
+    # FIXME mlpack@3.1.1: required but not yet available
+    # Ref: https://github.com/spack/spack/pull/38277
+    # depends_on("mlpack@3.1.1:", when="+mlpack")
+    conflicts("+mlpack", when="@25:", msg="mlpack@3.1.1: required but not yet available")
     depends_on("nlohmann-json @3.9.1:", when="@0.14: +json")
+    depends_on("podio @0.6", when="@25: +edm4hep")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts/package.py]:
```suggestion
    depends_on("podio @0.6:", when="@25: +edm4hep")
```

**Review by HadrienG2 — APPROVED** [review]
Thanks for the awesome continued support work!

**Review by becker33 — CHANGES_REQUESTED** [review]
Can you update this for mlpack now that #38277 is merged?

File: var/spack/repos/builtin/packages/acts/package.py [hunk 3]
```diff
@@ -271,12 +286,21 @@ class Acts(CMakePackage, CudaPackage):
     depends_on("hepmc3 @3.2.1:", when="+hepmc3")
     depends_on("heppdt", when="+hepmc3 @:4.0")
     depends_on("intel-tbb @2020.1:", when="+examples +tbb")
+    # FIXME mlpack@3.1.1: required but not yet available
+    # Ref: https://github.com/spack/spack/pull/38277
+    # depends_on("mlpack@3.1.1:", when="+mlpack")
+    conflicts("+mlpack", when="@25:", msg="mlpack@3.1.1: required but not yet available")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts/package.py]:
```suggestion
    depends_on("mlpack@3.1.1:", when="+mlpack")
```

**Review by tldahlgren — DISMISSED** [review]
Confirmed all new version commits, rest LGTM.

**Comment by wdconinc** [comment]
Thanks to @Zehvogel in #39189 for the 28.0.0 addition and ping to this PR. I've added the necessary hash here, as well as the newer version of actsvg required. Maybe this can get reviewed again and pushed?

It's over 3 months old and I don't think it includes any showstoppers. It would be nice to get `py-onnxruntime` working instead of adding a conflict, but that has turned out to be hardly trivial.

**Review by HadrienG2 — APPROVED** [review]

**Review by tldahlgren — COMMENTED** [review]
Re-confirmed version commits/sha256.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Comment by tldahlgren** [comment]
Looks like this PR cannot be merged until @becker33 releases his change request.

**Review by becker33 — APPROVED** [review]

**Pull request merged**